antiplag/models.py

```diff
@@ -41,42 +40,46 @@ class DocumentType(models.TextChoices):
     language = models.CharField(max_length=100, null=True)
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+    total_percentage = models.FloatField(null=True)
 
     def __str__(self):
         return f"{self.name if self.name else f'Document {self.id}'} ({self.type})"
 
-
-class Result(models.Model):
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    document = models.OneToOneField(
-        Document, on_delete=models.CASCADE, related_name="result"
-    )
-    matched_docs = models.JSONField(encoder=UUIDEncoder)
-    percentage = models.FloatField(default=0)
-
-    def __str__(self):
-        return f"{self.document}"
-
-    @property
-    def matches(self):
-        return len(self.matched_docs)
-
     @property
-    def intervals(self):
+    def ranges(self):
         indices = []
 
-        for matched_doc in self.matched_docs:
-            for match in matched_doc["intervals"]:
+        for result in self.results.all():
+            for range in result.ranges:
                 indices.append(
                     {
-                        "id": self.matched_docs.index(matched_doc),
-                        "name": matched_doc["name"],
-                        "percentage": matched_doc["percentage"],
-                        "from": match["fromA"],
-                        "to": match["toA"],
+                        "id": result.match_id,
+                        "name": result.match_name,
+                        "percentage": result.percentage,
+                        "from": range["fromA"],
+                        "to": range["toA"],
                     }
                 )
 
-        merged_intervals = merge_intervals(indices)
+        return merge_intervals(indices)
 
-        return merged_intervals
+
+class Result(models.Model):
+    class MatchType(models.TextChoices):
+        CORPUS = "CORPUS"
+        UPLOADED = "UPLOADED"
```

Comment:
Extract `MatchType` to `enums.py`.

Reply:
I would update attribute values to shorter, same length literals. What do you think?

Reply:
Agreed on moving. About shorting it - IMHO it's more clear in the DB, and 5 characters don't make such difference. I would rather not.

Reply:
OK, keep it as is.